warehouse/integrations/mssql/mssql.go

```diff
@@ -701,15 +701,17 @@ func (ms *MSSQL) loadUserTables(ctx context.Context) (errorMap map[string]error)
 		return
 	}
 
-	primaryKey := "id"
-	sqlStatement = fmt.Sprintf(`DELETE FROM %[1]s."%[2]s" FROM %[3]s _source where (_source.%[4]s = %[1]s.%[2]s.%[4]s)`, ms.Namespace, warehouseutils.UsersTable, ms.Namespace+"."+stagingTableName, primaryKey)
-	ms.logger.Infof("MSSQL: Dedup records for table:%s using staging table: %s\n", warehouseutils.UsersTable, sqlStatement)
-	_, err = tx.ExecContext(ctx, sqlStatement)
-	if err != nil {
-		ms.logger.Errorf("MSSQL: Error deleting from main table for dedup: %v\n", err)
-		_ = tx.Rollback()
-		errorMap[warehouseutils.UsersTable] = err
-		return
+	if ms.enableDedup() {
```

Comment:
We weren't checking any flags for dedup here, just pinging you guys to make sure it makes sense to do so before I add the tests. @lvrach @achettyiitr